primitives/stallable.futil

```diff
@@ -0,0 +1,14 @@
+extern "stallable.sv" {
+    // A latency-sensitive multiplier that takes 4 cycles to compute its result.
+    // If stall is set to a value of 1, the multiplier will stall until the value is
+    // set back to 0.
+    static<4> primitive stallable_mult[WIDTH] (
```

Comment:
Looks great! Does this need an `enable` signal? I'm wondering if there is a way to combine pipelined and stallable implementations by tying `stall` to zero for the former.

Reply:
An enable signal on this kind of thing is kinda meaningless because there can't be a done signal. The only reason to have an enable signal would be to prevent garbage values from being passed through the multiplier in hopes of reducing power utilization. And yes, we could combine these with pipelined primitives by tying the stall signal to zero since RTL synthesis should easily be able to optimize out the stall signal in that case, but I just want to reduce the complexity exposed to the user (i.e. they shouldn't see the stall signal in the interface if they don't need to use it). Any changes to the internal implementation is fine for me as long as this condition of users not seeing the stall signal if they just want to use a regular pipelined primitive holds.